.pre-commit-config.yaml

```diff
@@ -75,6 +75,8 @@ repos:
     hooks:
       - id: shfmt
         args: ["-i", "4"]
+      - id: shellcheck
+        files: deploy/azure/.*sh
```

Comment:
We can add this to other scripts but most of them fail so I've added it just for this script as an opportunity to use it in new scripts. We can track improvements on the rest on a new ticket.

This is IMO a good idea, especially for public-facing scripts that are run by users.